com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeBrickIndex.cs

```diff
@@ -49,26 +51,48 @@ struct ReservedBrick
             public int flattenedIdx;
         }
 
-        struct VoxelMeta
+        class VoxelMeta
         {
-            public RegId id;
-            public List<ushort> brickIndices;
+            public Cell cell;
+            public List<ushort> brickIndices = new List<ushort>();
+
+            public void Clear()
+            {
+                cell = null;
+                brickIndices.Clear();
+            }
         }
 
-        struct BrickMeta
+        class BrickMeta
         {
-            public HashSet<Vector3Int> voxels;
-            public List<ReservedBrick> bricks;
+            public HashSet<Vector3Int> voxels = new HashSet<Vector3Int>();
+            public List<ReservedBrick> bricks = new List<ReservedBrick>();
+
+            public void Clear()
+            {
+                voxels.Clear();
+                bricks.Clear();
+            }
         }
 
         Vector3Int m_CenterRS;   // the anchor in ref space, around which the index is defined. [IMPORTANT NOTE! For now we always have it at 0, so is not passed to the shader, but is kept here until development is active in case we find it useful]
 
         Dictionary<Vector3Int, List<VoxelMeta>> m_VoxelToBricks;
-        Dictionary<RegId, BrickMeta> m_BricksToVoxels;
+        Dictionary<Cell, BrickMeta> m_BricksToVoxels;
+
+        // Various pools for data re-usage
+        ObjectPool<BrickMeta> m_BrickMetaPool = new ObjectPool<BrickMeta>(x => x.Clear(), null, false);
+        ObjectPool<List<VoxelMeta>> m_VoxelMetaListPool = new ObjectPool<List<VoxelMeta>>(x => x.Clear(), null, false);
+        ObjectPool<VoxelMeta> m_VoxelMetaPool = new ObjectPool<VoxelMeta>(x => x.Clear(), null, false);
 
         int m_VoxelSubdivLevel = 3;
 
         bool m_NeedUpdateIndexComputeBuffer;
+        int m_UpdateMinIndex = int.MaxValue;
+        int m_UpdateMaxIndex = int.MinValue;
+
+        // Static variable required to avoid allocations inside lambda functions
+        static Cell g_Cell = null;
```

Comment:
yikes, can we use the cell index instead to do the checks?

Reply:
We can but I don't think that would make any difference, would it? It's just comparing references.

Reply:
would we still need to have the global static thing? I guess?

Reply:
Yeah, the static is mandatory because we use lambda for finding stuff. So it would capture the variable and generate gcalloc without it.
I guess we can make functors and set the ref in it to avoid this but it's much more verbose for not a lot of gain (I don't expect this to be run multi-threaded)